python/cuspatial/cuspatial/core/_column/geocolumn.py

```diff
@@ -130,10 +130,11 @@ def __init__(
             self.polygons.name = "polygons"
         else:
             raise TypeError("All four Tuple arguments must be cudf.ListSeries")
-        base = cudf.core.column.column.arange(0, len(self), dtype="int32").data
-        super().__init__(base, size=len(self), dtype="int32")
+        super().__init__(None, size=len(self), dtype="geometry")
         if shuffle_order is not None:
-            self._data = shuffle_order
+            self._gather_map = shuffle_order
```

Comment:
Can you just change the order of `type_codes` and `union_offset` instead of saving an extra level of indirection?